app/controllers/settings/registries_controller.rb

```diff
@@ -0,0 +1,103 @@
+# Settings::RegistriesController is responsibe to manage all the requests
+# related to the registries feature
+class Settings::RegistriesController < SettingsController
+  before_action :set_registry, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @registries = Registry.displayable
+  end
+
+  def new
+    @registry = Registry.new
+    @cert = Certificate.new
+  end
+
+  def create
+    @registry = Registry.new(registry_params.except(:certificate))
+    @cert = Certificate.find_or_initialize_by(certificate: certificate_param)
+
+    ActiveRecord::Base.transaction do
+      begin
+        @registry.save!
+        create_or_update_certificate! if certificate_param.present?
+        @created = true
+      rescue StandardError
+        raise ActiveRecord::Rollback
```

Comment:
There's no need to do this, `.transaction` will rollback if it gets any exception: https://github.com/rails/rails/blob/05eccb116ded8fd1250524cc4314907ff77511cd/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb#L231-L236

Reply:
I read that it wasn't necessary but without that I get `ActiveRecord::RecordInvalid` exception on my face all the time instead of just rolling back and having page rendered with errors as expected. Tests also fail because nobody catches this exception.

Do you have any idea why?

Reply:
We talked about that on IRC, but writing here for the sake of completion. `.transaction` will rollback in case of an exception and will bubble up the exception again, so we also need to catch it on our controller.